views/default/navigation/pagination.php

```diff
@@ -41,6 +44,21 @@
 	$base_url = current_page_url();
 }
 
+if ($url_fragment && !preg_match('~#.~', $base_url)) {
+	// we'll sometimes add fragment
+	$get_href = function ($offset) use ($base_url, $offset_key, $url_fragment) {
```

Comment:
1. A global function would require either injecting a bunch of state in arguments, or using global state. This keeps the use of the function cleaner: `$get_href($offset)`
2. I don't see how this would be reusable.
3. Combining into 1 anon function would be messier, but that could be a compromise to this.